### PR TITLE
feat: Add Gemini Exchange Adaptor

### PR DESCRIPTION
**Pull request by bowd**
## Description

This change adds an exchange adapter for Gemini

## Other changes

N/A

## Tested

- Added unit tests to verify logic works as expected
- Ran oracle locally to verify price retrieved from Gemini was correct

## Related issues

- Fixes https://github.com/mento-protocol/mento-general/issues/116

## Backwards compatibility

N/A

### CODE REVIEW

File: src/exchange_adapters/gemini.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+     */
+    return []
+  }
+
+  protected generatePairSymbol(): string {
+    return `${BaseExchangeAdapter.standardTokenSymbolMap.get(
+      this.config.baseCurrency
+    )}${BaseExchangeAdapter.standardTokenSymbolMap.get(
+      this.config.quoteCurrency
+    )}`.toLocaleLowerCase()
```
Comment (nvtaveras) [inline src/exchange_adapters/gemini.ts:31]: nit: how about separating these in variables to make it a bit more readable?

File: src/exchange_adapters/gemini.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  async fetchTicker(): Promise<Ticker> {
+    return this.parseTicker(
+      await this.fetchFromApi(ExchangeDataType.TICKER, `pubticker/${this.pairSymbol}`)
+    )
+  }
+
+  async fetchTrades(): Promise<Trade[]> {
+    /**
+     * Trades are cool, but empty arrays are cooler.
+     *                          @bogdan, 01/2023
```
Comment (nvtaveras) [inline src/exchange_adapters/gemini.ts:21]: 😎

File: src/exchange_adapters/gemini.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+    this.verifyTicker(ticker)
+    return ticker
+  }
+  /**
+   * No Gemini endpoint available to check for order book liveness.
+   * @returns bool
+   */
+  async isOrderbookLive(): Promise<boolean> {
+    return true
```
Comment (nvtaveras) [inline src/exchange_adapters/gemini.ts:74]: We can check the status of the orderbook using https://docs.gemini.com/rest-api/#symbol-details

**Review by nvtaveras — CHANGES_REQUESTED** [review]

File: src/exchange_adapters/gemini.ts (hunk 3 above)
Comment (bowd) [reply src/exchange_adapters/gemini.ts]: 👍 uff I missed this somehow, thanks!

File: src/exchange_adapters/gemini.ts (hunk 1 above)
Comment (bowd) [reply src/exchange_adapters/gemini.ts]: ✅

**Review by nvtaveras — APPROVED** [review]
LGTM 🚢

**Review by philbow61 — APPROVED** [review]
LGTM 🚢

**Pull request merged**